methods/incidentReports.coffee

```diff
@@ -21,15 +21,22 @@ Meteor.methods
     return newId
 
   # similar to editIncidentReport, but allows you to set a single field without changing any other existing fields.
-  updateIncidentReport: (incident) ->
+  updateIncidentReport: (incident, fieldsToRemove=[]) ->
     _id = incident._id
     delete incident._id
     user = Meteor.user()
     if not Roles.userIsInRole(user._id, ['admin'])
       throw new Meteor.Error("auth", "User does not have permission to edit incidents")
     incident.modifiedByUserId = user._id
     incident.modifiedByUserName = user.profile.name
-    res = Incidents.update({_id: _id}, {$set: incident})
+    updateOperators =
+      $set: incident
+    if fieldsToRemove.length
+      fieldsToUnset = {}
+      fieldsToRemove.forEach (field) ->
```

Comment:
Looks like using `updateIncidentReport` adds the new incident type property and leaves the existing. `editIncidentReport` will replace the entire object which is partially what we want to do but we loose several properties in the process (addedByUserName, addedByUserId, addedDate, etc.) Should we go this route and add those properties to the form-based incident object we pass to the method?